src/vs/workbench/workbench.web.api.ts

```diff
@@ -34,6 +34,26 @@ interface IExternalUriResolver {
 	(uri: URI): Promise<URI>;
 }
 
+interface TunnelOptions {
```

Comment:
TunnelOptions and Tunnel are copied from vscode.proposed.d.ts. ITunnelFactory is "inspired" by the tunnelFactory in vscode.proposed.d.ts.